lib/stages/VisSharedMemWriter.cpp

```diff
@@ -201,8 +220,8 @@ void VisSharedMemWriter::write_to_memory(const visFrameView& frame, uint32_t tim
 
     // notify that time_ind and freq_ind are being written to, by setting that
     // location to invalid in the access record
-    if (sem_wait(sem) == -1) {
-        FATAL_ERROR("Failed to acquire semaphore {}", _sem_name);
+    if (wait_for_semaphore() == false) {
+        DEBUG("Failed to acquire semaphore {}. Dropping frame.\n", _sem_name);
```

Comment:
If this happens, a reader is not releasing the semaphore. Higher log level? Also, should we publish a prometheus metric counting dropped frames here?

Reply:
Oh, I have not integrated prometheus metrics at all! That would be a good stretch-item to do.